### PR TITLE
Decouple resource attributes and zipkin span name from RumAttributeAp…

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
…pender

### CODE REVIEW

**Review by breedx-splk — APPROVED** [review]
Seems ok to me. I'd still like us to continue to consider how better to express which set of attributes belongs to which component, and how to explain the _why_. It's just not yet clear to me.

File: splunk-otel-android/src/test/java/com/splunk/rum/SplunkSpanDataModifierTest.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                                                                "com.example.Error")
+                                                        .put(
+                                                                SemanticAttributes
+                                                                        .EXCEPTION_MESSAGE,
+                                                                "failed")
+                                                        .put(
+                                                                SemanticAttributes
+                                                                        .EXCEPTION_STACKTRACE,
+                                                                "<stacktrace>")
+                                                        .build())))
```
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply splunk-otel-android/src/test/java/com/splunk/rum/SplunkSpanDataModifierTest.java:90]: AssertJ and nodejs levels of nesting function calls too

File: splunk-otel-android/src/main/java/com/splunk/rum/RumAttributeAppender.java [hunk 2]
```diff
@@ -39,49 +33,29 @@
...
-        span.setAttribute(RUM_VERSION_KEY, rumVersion);
-
-        span.setAttribute(DEVICE_MODEL_NAME, Build.MODEL);
-        span.setAttribute(DEVICE_MODEL_IDENTIFIER, Build.MODEL);
-        span.setAttribute(OS_NAME, "Android");
-        span.setAttribute(OS_TYPE, "linux");
-        span.setAttribute(OS_VERSION, Build.VERSION.RELEASE);
 
         String currentScreen = visibleScreenTracker.getCurrentlyVisibleScreen();
         span.setAttribute(SplunkRum.SCREEN_NAME_KEY, currentScreen);
```
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply splunk-otel-android/src/main/java/com/splunk/rum/RumAttributeAppender.java:57]: I think things might clarify by themselves in a few PRs -- I think that `RumAttributeAppender` (and the notion of "rum attributes") will disappear in a few PRs, and we'll end up only having attributes added by "OTel"/built-in instrumentations (e.g. network), which will be added in `SpanProcessor`s; and Splunk modifications, done in a custom `SpanExporter`.

**Comment by mateuszrzeszutek** [comment]
> Seems ok to me. I'd still like us to continue to consider how better to express which set of attributes belongs to which component, and how to explain the _why_. It's just not yet clear to me.

I think this might become clear when we start moving things to OTel package; I'm planning to split/untangle lots of the spaghetti that we have right now around the initialization process.

**Pull request merged**